subprojects/platform-native/src/main/java/org/gradle/nativeplatform/tasks/InstallExecutable.java

```diff
@@ -184,13 +199,28 @@ protected FileOperations getFileOperations() {
 
     @TaskAction
     public void install() {
+        deleteDestinationDir();
+
+        if (!getExecutable().exists()) {
+            setDidWork(false);
```

Comment:
I wonder how setting `didWork` to `false` but actually having deleted the destination directory might affect caching logic. In other words, "But, you actually did do work. You deleted a directory."